src/components/FormItem/index.tsx

```diff
@@ -0,0 +1,143 @@
+import { Form, Select, FormItemProps as antFormItemProps, TimePicker, Flex, InputNumber } from 'antd';
+import { Input } from 'aelf-design';
+import { memo } from 'react';
+import FormGroup from './components/FormGroup';
+import FormTree from './components/FormTree';
+import Upload from '../AWSUpload';
+import FormDatePiker from './components/FormDatePicker';
+import grayArrow from 'assets/images/project/grayArrow.svg';
+
+import {
+  datePickerProps,
+  rowProps,
+  fileUploadProps,
+  groupProps,
+  idCardUploadProps,
+  inputProps,
+  passwordProps,
+  selectProps,
+  textAreaProps,
+  treeProps,
+  searchSelectProps,
+  timePickerProps,
+  customizeProps,
+  inputNumberProps,
+  inlineFieldProps,
+  FieldsGroupProps,
+  pureTextProps,
+} from './types';
+export type FormItemProps = (
+  | inputProps
+  | textAreaProps
+  | groupProps
+  | treeProps
+  | passwordProps
+  | selectProps
+  | datePickerProps
+  | fileUploadProps
+  | idCardUploadProps
+  | rowProps
+  | searchSelectProps
+  | timePickerProps
+  | customizeProps
+  | inputNumberProps
+  | inlineFieldProps
+  | pureTextProps
+  | FieldsGroupProps
+) &
+  antFormItemProps;
+
+function getChildren(type: FormItemProps['type'], childrenProps: FormItemProps['childrenProps']) {
```

Comment:
so many cases, please use object.

Reply:
It will be improved in the next version.